app/src/test/java/com/dbychkov/data/datastore/DatabaseFlashcardDataStoreTest.java

```diff
@@ -0,0 +1,219 @@
+package com.dbychkov.data.datastore;
+
+import com.dbychkov.data.greendao.DaoSession;
+import com.dbychkov.data.greendao.FlashcardEntity;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+
+/**
+ * Created by brian on 5/6/16.
+ * Its very important to test the data access layer to make sure that CRUD ops are done correctly
+ * and data integrity is preserved
+ * This class tests the DatabaseFlashcardDataStore but DatabaseLessonDataStore,
+ * FlashcardDataRepository and LessonDataRepository should be tested in very similar way
+ */
+public class DatabaseFlashcardDataStoreTest {
+
+    DatabaseFlashcardDataStore databaseFlashcardDataStore;
+
+    /**
+     * init a databaseflashcarddatastore with injected mock object
+     *
+     * @throws Exception
+     */
+    @Before
+    public void setUp() throws Exception {
+        databaseFlashcardDataStore = new DatabaseFlashcardDataStore(mock(DaoSession.class));
+    }
+
+    /**
+     * init a new databaseflashcardatastore
+     *
+     * @throws Exception
+     */
+    @After
+    public void tearDown() throws Exception {
+        databaseFlashcardDataStore = new DatabaseFlashcardDataStore(mock(DaoSession.class));
+    }
+
+
+    @Test
+    public void testAddFlashcard() throws Exception {
+        FlashcardEntity flashcardEntity = new FlashcardEntity(1L);
+        databaseFlashcardDataStore.addFlashcard(flashcardEntity);
+        FlashcardEntity result =
+                databaseFlashcardDataStore.getFlashcardById(1L).toBlocking().singleOrDefault(null);
+        assertEquals(flashcardEntity, result);
+    }
+
+    @Test
+    public void testRemoveFlashcard() throws Exception {
+        FlashcardEntity flashcardEntity = new FlashcardEntity(1L);
+        databaseFlashcardDataStore.addFlashcard(flashcardEntity);
+        FlashcardEntity result =
+                databaseFlashcardDataStore.getFlashcardById(1L).toBlocking().singleOrDefault(null);
+        if (result == null) {
```

Comment:
You are right, an assertNotEqual statement would be much better here. The reason why I used an if statement at the time was because using a fail() allowed me to provide a customized failure message for debugging purposes. I've never used fail() before and just wanted to see how it works.